packages/camera/camera/example/ios/RunnerTests/CameraDispatchQueueRaceConditionTests.m

```diff
@@ -0,0 +1,41 @@
+//
+//  CameraDispatchQueueRaceConditionTests.m
+//  RunnerTests
+//
+//  Created by Huan Lin on 12/16/21.
+//  Copyright © 2021 The Flutter Authors. All rights reserved.
+//
+
+@import camera;
+@import camera.Test;
+@import XCTest;
+
+@interface CameraDispatchQueueRaceConditionTests : XCTestCase
+@end
+
+@implementation CameraDispatchQueueRaceConditionTests
+
+- (void)testRaceCondition {
+  CameraPlugin *camera = [[CameraPlugin alloc] initWithRegistry:nil messenger:nil];
+
+  XCTestExpectation *disposeExpectation =
+      [[XCTestExpectation alloc] initWithDescription:@"dispose's result must be called"];
+  FlutterMethodCall *disposeCall = [FlutterMethodCall methodCallWithMethodName:@"dispose"
+                                                                     arguments:nil];
+  FlutterMethodCall *createCall = [FlutterMethodCall
+      methodCallWithMethodName:@"create"
+                     arguments:@{@"resolutionPreset" : @"medium", @"enableAudio" : @(1)}];
+  // Mimic a dispose call followed by a create call, which can be triggered by slightly dragging the
+  // home bar, causing the app to be inactive, and immediately regain active.
+  [camera handleMethodCall:disposeCall
+                    result:^(id _Nullable result) {
+                      [disposeExpectation fulfill];
+                    }];
+  [camera handleMethodCall:createCall
+                    result:^(id _Nullable result){
+                    }];
```

Comment:
The old code might also pass this test (since it was a race condition), right?
Maybe we don't even need to call `createCall` here and make the test more like:
"dispatchQueue should not be nil after dispose"

Reply:
The old code fails the test. I added `createCall` immediately following `dispose` to illustrate the race, otherwise a single `dispose` was ok to nil out the reference.